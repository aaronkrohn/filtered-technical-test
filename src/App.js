import React from 'react';
import styled from 'styled-components';
import Sidebar from './containers/Sidebar';
import Preview from './containers/Preview';
import Header from './containers/Header';

const Wrapper = styled.div`
  padding: 3.125em 0;
`;

const MainWrapper = styled.main`
  margin: 3.125em auto;
  max-width: 71.25em;
  padding: 0 1em;
`;

const AssetsView = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 -1em;
`;

const App = () => (
  <Wrapper>
    <Header />
    <MainWrapper>
      <AssetsView>
        <Sidebar />
        <Preview />
      </AssetsView>
    </MainWrapper>
  </Wrapper>
);

export default App;
