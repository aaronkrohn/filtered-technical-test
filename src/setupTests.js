import React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from './reducers/index';

export const renderWithRedux = (
  ui,
  { initialState, store = createStore(reducers, initialState) } = {}
) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store
});
