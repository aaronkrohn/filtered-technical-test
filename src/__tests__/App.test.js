import React from 'react';
import { renderWithRedux } from '../setupTests';
import App from '../App';

test('renders', () => {
  const { container } = renderWithRedux(<App />);
  expect(container.firstChild).toMatchSnapshot();
});
