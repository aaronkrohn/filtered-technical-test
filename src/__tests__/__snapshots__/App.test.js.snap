// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  class="wrapper"
>
  <div
    class="asset-summary"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-6 asset-summary-left"
        >
          <span>
            4 assets
          </span>
        </div>
        <div
          class="col-6 asset-summary-right"
        >
          2x
          <span
            class="info"
          >
            Articles
          </span>
          1x
          <span
            class="info"
          >
            Documents
          </span>
          1x
          <span
            class="info"
          >
            Video
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container asset-builder"
  >
    <div
      class="row"
    >
      <div
        class="col-4"
      >
        <ul
          class="asset-picker"
        >
          <li
            class="asset type-video"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Video
              </span>
            </p>
          </li>
          <li
            class="asset type-video"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Video
              </span>
            </p>
          </li>
          <li
            class="asset type-video"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Video
              </span>
            </p>
          </li>
          <li
            class="asset type-video"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Video
              </span>
            </p>
          </li>
        </ul>
      </div>
      <div
        class="col-8"
      >
        <h2>
          Learning Assets Preview
        </h2>
        <ul
          class="asset-preview"
        >
          <li
            class="asset type-article"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Articles
              </span>
            </p>
            <button
              class="remove-asset"
            >
              x
            </button>
          </li>
          <li
            class="asset type-article"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Articles
              </span>
            </p>
            <button
              class="remove-asset"
            >
              x
            </button>
          </li>
          <li
            class="asset type-document"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info "
              >
                Document
              </span>
            </p>
            <button
              class="remove-asset"
            >
              x
            </button>
          </li>
          <li
            class="asset type-video"
          >
            <h2>
              Dummy asset name
            </h2>
            <p>
              <span
                class="info"
              >
                4 minutes
              </span>
              <span
                class="info"
              >
                Video
              </span>
            </p>
            <button
              class="remove-asset"
            >
              x
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
